mobile_back_apple/cpp/backend_coreml/coreml_util.swift

```diff
@@ -117,18 +116,21 @@ private class MLCommonsComputationUnit {
   {
     switch computationUnitName {
     case .all:
-      return MLComputeUnits.all
+      if let hasANE = Device.hasANE, hasANE {
+        return MLComputeUnits.all
+      } else {
+        return MLComputeUnits.cpuAndGPU
+      }
     case .cpuAndGPU:
       return MLComputeUnits.cpuAndGPU
     case .cpuAndNeuralEngine:
-      if #available(iOS 16.0, *) {
+      if let hasANE = Device.hasANE, hasANE, #available(iOS 16.0, macOS 13.0, *) {
         return MLComputeUnits.cpuAndNeuralEngine
+      } else {
```

Comment:
We can refactor here slightly, and remove the else. So 
if (expression) { 
    return value
}
return anotherValue

Reply:
I prefer to have the else here to quickly scan the code and see that there are 2 possible exists.